sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -630,6 +652,11 @@ && getConnectionInitSqls().get() != null
         if (getDriverClassLoader() != null) {
           basicDataSource.setDriverClassLoader(getDriverClassLoader());
         }
+        if (getDriverJars() != null) {
+            URLClassLoader classLoader =
```

Comment:
I'm not sure I understand your question entirely.

These jars should be staged in each worker when the datasource is built

Reply:
Java dependencies are usually added via the filesToStage option [1] (which defaults to everything in CLASSPATH). Just specifying a jar here does not result in the jar being available to workers. So users will also need to make sure the add these jars to the filesToStage option or the CLASSPATH in-addition to providing here. Is that correct ?

[1] https://github.com/apache/beam/blob/7134cfdd9daffa649c5401c79cc321732302c53b/sdks/java/core/src/main/java/org/apache/beam/sdk/options/FileStagingOptions.java#L36

Reply:
If so, please update the documentation to mention this.

Reply:
No, passing in a jar here should be enough. Users will not need to have the jar in the classpath or add the jar in `filesToStage`.

`driverJars` parameter should be used in the case where the driver jar is not available in the classpath. Jars passed in using this parameter will be localized and passed in using `basicDataSource.setDriverClassLoader()`, which will then be used to load the JDBC driver. [1]

This change will be helpful for templates, since we want the template to run with any database, and adding all possible database drivers is infeasible. We have done this in google-provided templates by creating a copy of JdbcIO called DynamicJdbcIO [2].

[1] https://commons.apache.org/proper/commons-dbcp/apidocs/org/apache/commons/dbcp2/BasicDataSource.html#setDriverClassLoader-java.lang.ClassLoader- 
[2] https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v1/src/main/java/com/google/cloud/teleport/io/DynamicJdbcIO.java

Reply:
I see. Seems like the jar will be basically re-downloaded in workers when creating the source.